src/Acquisition/_Acquisition.c

```diff
@@ -970,6 +972,20 @@ Wrapper_unicode(Wrapper *self)
     }
 }
 
+static PyObject *
+Wrapper_bytes(Wrapper *self)
+{
+    PyObject *r;
+
+    if ((r = PyObject_GetAttr(OBJECT(self), py__bytes__))) {
+        ASSIGN(r, PyObject_CallFunction(r, NULL, NULL));
+        return r;
+    } else {
+        PyErr_Clear();
+        return Wrapper_str(self);
```

Comment:
It's not clear to me that the 'no such attribute' error case is correct, in either the C or Python versions. 

For `repr`, in both implementations, if the lookup fails, we simply return `repr(aq_inner(self))` (aka `self->obj`) which goes through some complicated CPython internal logic. All good, won't fail.

For `__unicode__` (which only matters on Python 2), the C implementation catches the error and defers to the `str()` implementation, which is what CPython does implicitly for objects implemented in Python, I think (which is why the Python implementation doesn't do anything special).

For `__bytes__`, the C implementation will catch an error and instead call `str(self)`. That *might* work on Python 2, where str-is-bytes, but on Python 3, where str-is-unicode, it might fail, not least because of `str` having non-ASCII characters. Does CPython ordinarily do that automatically in Python 3? Let's check:

```pycon
$ python3.7
Python 3.7.3 (default, Mar 30 2019, 03:37:43)
[Clang 10.0.0 (clang-1000.11.45.5)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> class WithBytes:
...     def __bytes__(self):
...         return b'my bytes'
...     def __str__(self):
...         return u'my snowman: \u2603'
...
>>> bytes(WithBytes())
b'my bytes'
>>> class WithoutBytes:
...     def __str__(self):
...         return u'my snowman: \u2603'
...
>>> bytes(WithoutBytes())
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: cannot convert 'WithoutBytes' object to bytes
```

So if there's no `__bytes__` method, CPython raises an error. To do the same, the `else` case here should probably be something like `return PyBytes_FromObject(self->obj)` (I'm not 100% sure that's the right call).

Having explored that, it looks like the pure python implementation is doing the right thing.

Either way, it needs some tests for the negative case.